README.md
#[Stanford Video](https://github.com/SU-SWS/stanford_video)
##### Version: 7.x-2.3

Maintainers: [jbickar](https://github.com/jbickar), [sherakama](https://github.com/sherakama)
[Changelog.txt](CHANGELOG.txt)

This module creates a custom content type that uses the JWPlayer video player for playing streaming video hosted by Stanford Video (http://stanfordvideo.stanford.edu/streaming), and integrating captions provided by the Stanford Office of Accessible Education's Caption Tool (http://captioning.stanford.edu).

Installation
---

Install as you would with any other Drupal contributed module. This module requires the JWPlayer Javascript library; download it from http://www.longtailvideo.com/jw-player/download/ and place it in sites/all/libraries.

Configuration
---

* Create a new video node at node/add/stanford-video
* You can upload videos directly (.flv, .mp3, .mp4, .m4v, .webm) and attach captions (.srt, .dfxp, .xml)
* You also can stream videos from an RTMP streamer; the URL for the Stanford Video streaming server is provided by default
** For more information on streaming video hosting through Stanford Video, visit http://stanfordvideo.stanford.edu/streaming
** For more information on creating caption files for videos, visit http://captioning.stanford.edu
* You also can embed videos from third-party providers (YouTube, Vimeo, etc.) by pasting the embed code into the "External Video Embed" field
** This field relies on the "Full HTML" text format. You should only grant access to that text format to trusted users and roles.

Troubleshooting
---

If you are experiencing issues with this module try reverting the feature first. If you are still experiencing issues try posting an issue on the GitHub issues page.

Contribution / Collaboration
---

You are welcome to contribute functionality, bug fixes, or documentation to this module. If you would like to suggest a fix or new functionality you may add a new issue to the GitHub issue queue or you may fork this repository and submit a pull request. For more help please see [GitHub's article on fork, branch, and pull requests](https://help.github.com/articles/using-pull-requests)


KNOWN ISSUES
------------
The JW Player captions plugin is not compatible with the Firefox "HTTPS Everywhere" plugin.
Captions will not display if you have this Firefox plugin enabled.


CHANGELOG.txt
Stanford Web Video x.x-x.x xxxx-xx-xx
---------------------------------

7.x-2.3                           2018-06-28
--------------------------------------------
- Added composer.json

7.x-2.2+6-dev                     2017-01-20
--------------------------------------------
- Set content type to published by default

7.x-2.2                           2016-05-30
--------------------------------------------
- VPSADEV-136: remove features exclude reference to vpsa_landing_page
- VPSADEV-146: Added author exposed filter and view row counter to manage content view
- VPSADEV-146: Reinstated VBO Actions
- Re-export Feature to be compatible with newer versions of Features.
- jbickar Fixing up the VBO

7.x-2.1                           2015-03-19
--------------------------------------------
- BASIC-770 | Add manage video view and link to site actions menu
- Removed shortcuts dependency
- Removed self dependency and added fe_exclude to body base
- Removed menu links into jsvpsa
- Update stanford_video.info
- Update and rename README.txt to README.md


Stanford Web Video 7.x-2.0-alpha2 2014-03-31
--------------------------------------------
Updated to be compatible with Features 2.x

Stanford Web Video 7.x-1.0-beta1 2013-06-27
-------------------------------------------
Added instructions for working with videos in subdirectories on Stanford Video
  streaming service (thanks to Don Davis of Stanford Video)

Stanford Web Video 7.x-1.0-alpha7 2013-03-25
--------------------------------------------
Added help at admin/help/stanford_video

Stanford Web Video 7.x-1.0-alpha6 2013-03-20
--------------------------------------------
Added View (stanford_web_videos)
Fixed HTML5/streaming conflict issue
Added 320x240 JWPlayer preset

Stanford Web Video 7.x-1.0-alpha5 2013-03-19
----------------------------------------
Rearranged fields in Stanford Web Video content type
Bug fixes

Stanford Web Video 7.x-1.0-alpha4 2013-03-19
----------------------------------------
Added support for local (uploaded) video and embedded video from external providers.

Stanford Web Video 7.x-1.0-alpha2 2013-03-13
----------------------------------------
Initial working release. Supports RTMP streaming video from the Stanford Web Video service.

Stanford Web Video 7.x-1.0-alpha1 2012-09-17
----------------------------------------
Initial, incomplete release


node-stanford_video.tpl.php
<?php 
$allowed_tags = array('img', 'p', 'a', 'em', 'strong', 'ul', 'ol', 'li', 'dl', 'dt', 'dd', 'div');
$media = filter_xss($node->field_stanford_video_media[0]['filepath'], $allowed_tags);
$caption = filter_xss($node->field_stanford_video_caption[0]['filepath'], $allowed_tags);
$remote = filter_xss($node->field_stanford_video_remote[0]['view'], $allowed_tags);
$videocols = filter_xss($node->field_stanford_video_resolution[0]['view'], $allowed_tags);
$keyframe = filter_xss($node->field_stanford_video_keyframe[0]['filepath'], $allowed_tags);
$basepath = base_path();
$filepath = file_directory_path();
//kpr($node);
?>
 <script type="text/javascript" src="<?php print $basepath.$filepath; ?>/media/swfobject.js"></script>
<script type="text/javascript">
	swfobject.registerObject("player","9.0.98","expressInstall.swf");
</script>

<?php

if ($videocols == '480 X 320'){
    $output = "<div style=\"width: 480px; margin:auto; margin-bottom:2em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"480\" height=\"340\"><param name=\"movie\" value=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }
  elseif ($videocols == '320 X 240'){
    $output = "<div style=\"width: 320px; float:right; margin-left:1em; margin-right:1em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"320\" height=\"240\"><param name=\"movie\" value=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }			
  elseif ($videocols == '640 X 360'){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"640\" height=\"360\"><param name=\"movie\" value=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }	
  elseif ($videocols == '640 X 480'){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"640\" height=\"480\"><param name=\"movie\" value=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }				
  else {
    $output = t("I'm sorry, something has gone terribly wrong. ");
    $output .= l(t("Please contact the webmaster. "), "contact");
  }
  echo $output;
  print"<param name=\"flashvars\" value=\"file=";

  if ($remote != null){
    print $remote;
  }
  else {
    print $basepath.$media;
  }
if ($keyframe !=null){
		print "&amp;image=http://studentaffairs.stanford.edu/";print $keyframe;print"&amp;captions=";}		
else {			
	print"&amp;image=http://captioning.stanford.edu/images/startimage.png&amp;captions=";}
print $basepath.$caption;
print"\" />";

  if ($videocols == '480 X 320'){
    $output = "<div style=\"width: 480px; margin:auto; margin-bottom:2em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" width=\"480\" height=\"340\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }
  elseif ($videocols == '320 X 240'){
    $output = "<div style=\"width: 320px; float:right; margin-left:1em; margin-right:1em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" width=\"320\" height=\"240\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }			
  elseif ($videocols == '640 X 360'){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" width=\"640\" height=\"360\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }	
  elseif ($videocols == '640 X 480'){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $basepath.$filepath;
    $output .= "/media/player.swf\" width=\"640\" height=\"480\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }				
  else {
    $output = t("I'm sorry, something has gone terribly wrong. ");
    $output .= l(t("Please contact the webmaster."), "contact");
  }
  echo $output;

print "<param name=\"flashvars\" value=\"file=";
if ($remote != null){
		print $remote; }
else {
		print $basepath.$media;
	}
if ($keyframe !=null){
		print "&amp;image=".$basepath.$keyframe."&amp;captions=";
	}		
else {			
	print"&amp;image=http://studentaffairs.stanford.edu/sites/default/files/media/poster.png&amp;captions=";}
print $basepath.$caption;
print "\" />"; 

print"<p><a href=\"http://get.adobe.com/flashplayer\">Get the Adobe Flash Player</a> to view this presentation.</p></object></div></object></div>"; 

print $node->content['body']['#value'];

 ?>

<br class="clear" />


theme/jw_player-forked.tpl.php
<?php
/**
 * @file
 * Display the JW Player.
 *
 * Variables available:
 * - $html_id: Unique id generated for each video.
 * - $width: Width of the video player.
 * - $height: Height of the video player.
 * - $sources: An array of files to be played.
 * - $jw_player_inline_js_code: JSON data with configuration settings for the video player.
 * - $image: URL to an image to be used for the poster (ie. preview image) for this video.
 *
 * @see template_preprocess_jw_player()
 */
?>
<div class="jwplayer-video">
  <video id="<?php print $html_id ?>" width="<?php print $width ?>" height="<?php print $height ?>" controls="controls" preload="none"<?php if(isset($image)) : ?> poster="<?php print $image ?>"<?php endif ?>>
  <?php foreach ($sources as $source) { ?>
    <source src="<?php print $source['file_path']; ?>"<?php if (isset($source['file_mime'])): ?> type="<?php print $source['file_mime'] ?>"<?php endif ?> />
  <?php };?>
  </video>
</div>
<?php if(isset($jw_player_inline_js_code)): ?>
  <script type="text/javascript">
    jwplayer('<?php print $html_id ?>').setup(<?php print $jw_player_inline_js_code?>);
  </script>
<?php endif ?>


theme/stanford_jw_player.tpl.php
<?php
/**
 * @file
 * Display the JW Player.
 *
 * Variables available:
 * - $html_id: Unique id generated for each video.
 * - $width: Width of the video player.
 * - $height: Height of the video player.
 * - $file_url: The url of the file to be played.
 * - $jw_player_inline_js_code: JSON data with configuration settings for the video player.
 * - $poster: URL to an image to be used for the poster (ie. preview image) for this video.
 *
 * @see template_preprocess_jw_player()
 */
?>
<div class="jwplayer-video">
  <video id="<?php print $html_id ?>" width="<?php print $width ?>" height="<?php print $height ?>" controls="controls" preload="none"<?php if(isset($poster)) : ?> poster="<?php print $poster ?>"<?php endif ?>>
  <source src="<?php print $file_url ?>"<?php if (isset($file_mime)): ?> type="<?php print $file_mime ?>"<?php endif ?> />
  </video>
</div>
<?php if(isset($jw_player_inline_js_code)): ?>
  <script type="text/javascript">
    jwplayer('<?php print $html_id ?>').setup(<?php print $jw_player_inline_js_code?>);
  </script>
<?php endif ?>


content-field-field_stanford_video_media.tpl.php
<?php
// $Id: content-field.tpl.php,v 1.1.2.6 2009/09/11 09:20:37 markuspetrux Exp $

/**
 * @file content-field.tpl.php
 * Default theme implementation to display the value of a field.
 *
 * Available variables:
 * - $node: The node object.
 * - $field: The field array.
 * - $items: An array of values for each item in the field array.
 * - $teaser: Whether this is displayed as a teaser.
 * - $page: Whether this is displayed as a page.
 * - $field_name: The field name.
 * - $field_type: The field type.
 * - $field_name_css: The css-compatible field name.
 * - $field_type_css: The css-compatible field type.
 * - $label: The item label.
 * - $label_display: Position of label display, inline, above, or hidden.
 * - $field_empty: Whether the field has any valid value.
 *
 * Each $item in $items contains:
 * - 'view' - the themed view for that item
 *
 * @see template_preprocess_content_field()
 */

// Define allowed tags
$allowed_tags = array('img', 'p', 'a', 'em', 'strong', 'ul', 'ol', 'li', 'dl', 'dt', 'dd', 'div');

// video upload file (.flv, .mp3, .mp4, .m4v)
$media = filter_xss($node->field_stanford_video_media[0]['filepath'], $allowed_tags);

// caption file (.srt, .xml)
$caption = filter_xss($node->field_stanford_video_caption[0]['filepath'], $allowed_tags);

// remote streaming URL
$remote = filter_xss($node->field_stanford_video_remote[0]['safe'], $allowed_tags);

/**
 * CCK allowed values text field, set up in key/value pairs.
 * 1|480x340, 2|320x240, 3|640x360, 4|640x480
 */
$video_resolution = filter_xss($node->field_stanford_video_resolution[0]['value'], $allowed_tags);

// Keyframe upload file; defaults to captioning service .png
$keyframe = filter_xss($node->field_stanford_video_keyframe[0]['filepath'], $allowed_tags);
$keyframe_default = "http://captioning.stanford.edu/images/startimage.png";

// Drupal base path
$basepath = base_path();

// Drupal file directory (e.g., sites/default/files)
$filepath = file_directory_path();

// Module path
$stanford_video_path = url(drupal_get_path('module', 'stanford_video'));

// Name of the file at the remote destination, e.g., "media.flv," "media.mp4," etc.
preg_match('/[^\/]*.(flv|mp4|m4v|mp3)/', $remote, $matches);
$remote_file = $matches[0];

// URL to the streaming server group space, e.g., rtmp://sv-stream.stanford.edu/groupname
$splits = preg_split('/\/[^\/]*.(flv|mp4|m4v|mp3)/', $remote);
$remote_streamer = $splits[0];

drupal_add_js(drupal_get_path('module', 'stanford_video') . '/media/jwplayer.js');
//drupal_set_html_head('<meta http-equiv="Content-Type" content="video/mp4" />');

// Array of possible video window sizes to pass to HTML5 video tag
$video_window_size = array();
$video_window_size[1] = 'width="480" height="340"' . "\n";
$video_window_size[2] = 'width="320" height="240"' . "\n";
$video_window_size[3] = 'width="640" height="360"' . "\n";
$video_window_size[4] = 'width="640" height="480"' . "\n";

// Array of possible video window sizes to pass to JW Player
$jw_video_window_size = array();
$jw_video_window_size[1] = 'width: "480", height: "340",';
$jw_video_window_size[2] = 'width: "320", height: "240",';
$jw_video_window_size[3] = 'width: "640", height: "360",';
$jw_video_window_size[4] = 'width: "640", height: "480",';


/**
 * 
 * REMOTE VIDEO
 *
 * Generate the HTML and Javascript to serve a remote streaming video with JW Player.
 * This method relies strictly on Flash and therefore does not have an HTML5 fallback.
 */

//HTML
$remote_output = '<div id="stanford-video-container">You will need to enable Flash to view this video</div>';

//Javascript
$remote_output .= "\n" . '<script type="text/javascript">';
$remote_output .= "\n\t" . 'jwplayer("stanford-video-container").setup({';
$remote_output .= "\n\t\t" . "flashplayer: \"" . $stanford_video_path . "/media/player.swf\",";

// Check for keyframe; set it to the default if one doesn't exist.
if(!empty($keyframe)) {
  $remote_output .= "\n\t\t" . "image: \"" . $basepath.$keyframe . "\",";
} 
else {
  $remote_output .= "\n\t\t" . "image: \"" . $keyframe_default . "\",";
}
// Set the window size.
$remote_output .= "\n\t\t" . $jw_video_window_size[$video_resolution];
// RTMP provider
$remote_output .= "\n\t\t" . "provider: \"rtmp\",";
// RTMP streaming application location
$remote_output .= "\n\t\t" . "streamer: \"" . $remote_streamer . "\",";
// Remote file name
$remote_output .= "\n\t\t" . "file: \"" . $remote_file . "\",";
// Caption plugin
$remote_output .= "\n\t\t" . "'plugins': {\n\t\t\t'captions-2': {\n\t\t\t\t 'file': \"" . $basepath.$caption . "\"\n\t\t\t}\n\t\t}";
$remote_output .= "\n\t});"; //close the setup
$remote_output .= "\n" . '</script>';

/**
 * 
 * LOCAL VIDEO
 *
 * Generate the HTML and Javascript to serve a locally-hosted video with JW Player.
 * This method uses Flash first and has an HTML5 fallback.
 */

// HTML5 <video> tag, id and media src
$local_output = '<video id="stanford-video-container" src="' . $basepath. $media .'" ';
// Set the window size
$local_output .= $video_window_size[$video_resolution];
// Check for keyframe; set it to the default if one doesn't exist.
if(!empty($keyframe)) {
  $local_output .= 'poster="' . $basepath.$keyframe . '" ';
} 
else {
  $local_output .= 'poster="' . $keyframe_default . '" ';
}
$local_output .= ">You will need to enable Flash to view this video</video>";

//Javascript
$local_output .= "\n" . '<script type="text/javascript">';
$local_output .= "\n\t" . 'jwplayer("stanford-video-container").setup({';
$local_output .= "\n\t\t" . "flashplayer: \"" . $stanford_video_path . "/media/player.swf\",";

// Caption plugin
$local_output .= "\n\t\t" . "'plugins': {\n\t\t\t'captions-2': {\n\t\t\t\t 'file': \"" . $basepath.$caption . "\"\n\t\t\t}\n\t\t}";
$local_output .= "\n\t});"; //close the setup
$local_output .= "\n" . '</script>';




/**
 * Check for remote or local media.
 * If it's remote, use a regular div and use JW Player to serve it up.
 * If it's local, use a <video> tag and use JW Player to serve it up.
 */

// Check for remote first, and output the HTML/JS to serve that.
if(!empty($remote)) {
  print($remote_output);
}
// Check for local and output its HTML/JS.
elseif(!empty($media)) { 
  print($local_output);
}
// If that didn't work, return a warning.
else {
  drupal_set_message(t("Could not find a video file."), 'warning');
}


/*
if ($videocols == 1){
    $output = "<div style=\"width: 480px; margin:auto; margin-bottom:2em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"480\" height=\"340\"><param name=\"movie\" value=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }
  elseif ($videocols == 2){
    $output = "<div style=\"width: 320px; margin-left:1em; margin-right:1em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"320\" height=\"240\"><param name=\"movie\" value=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }			
  elseif ($videocols == 3){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"640\" height=\"360\"><param name=\"movie\" value=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }	
  elseif ($videocols == 4){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object id=\"player\" classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\" name=\"player\" width=\"640\" height=\"480\"><param name=\"movie\" value=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }				
  else {
    $output = t("I'm sorry, something has gone terribly wrong. ");
    $output .= l(t("Please contact the webmaster. "), "contact");
  }
  echo $output;
  print"<param name=\"flashvars\" value=\"file=";

  if ($remote != null){
    print $remote_file ."&amp;streamer=". $remote_streamer;
  }
  else {
    print $basepath.$media;
  }
if ($keyframe !=null){
		print "&amp;image=http://studentaffairs.stanford.edu/";print $keyframe;print"&amp;captions=";}		
else {			
	print"&amp;image=http://captioning.stanford.edu/images/startimage.png&amp;captions=";}
print $basepath.$caption;
print"\" />";

  if ($videocols == 1){
    $output = "<div style=\"width: 480px; margin:auto; margin-bottom:2em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" width=\"480\" height=\"340\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }
  elseif ($videocols == 2){
    $output = "<div style=\"width: 320px; margin-left:1em; margin-right:1em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" width=\"320\" height=\"240\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }			
  elseif ($videocols == 3){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" width=\"640\" height=\"360\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }	
  elseif ($videocols == 4){
    $output = "<div style=\"width: 640px; margin:auto; margin-bottom:2em;\"><object type=\"application/x-shockwave-flash\" data=\"";
    $output .= $stanford_video_path;
    $output .= "/media/player.swf\" width=\"640\" height=\"480\"><param name=\"movie\" value=\"player.swf\" /><param name=\"allowfullscreen\" value=\"true\" /><param name=\"allowscriptaccess\" value=\"always\" />";
  }				
  else {
    $output = t("I'm sorry, something has gone terribly wrong. ");
    $output .= l(t("Please contact the webmaster."), "contact");
  }
  echo $output;

print "<param name=\"flashvars\" value=\"file=";
if ($remote != null){
    print $remote_file ."&amp;streamer=". $remote_streamer;
}
else {
		print $basepath.$media;
	}
if ($keyframe !=null){
		print "&amp;image=".$basepath.$keyframe."&amp;captions=";
	}		
else {			
	print"&amp;image=http://studentaffairs.stanford.edu/sites/default/files/media/poster.png&amp;captions=";}
print $basepath.$caption;
print "\" />"; 

print"<p><a href=\"http://get.adobe.com/flashplayer\">Get the Adobe Flash Player</a> to view this presentation.</p></object></div></object></div>"; 

*/
 ?>

<br class="clear" />
